### DISCUSSION TITLE
Skip Middleware?

### DISCUSSION THREAD

**Discussion by bruceharrison1984** [honojs/hono#1498] [General]
I there anyway to say that a particular path (ie `/static/`) shouldn't have middleware run against it? The only working solution I've found is to declare every route except the one that I want to be skipped.

In the case of static assets with CF pages, middleware trumps the route and prevents the browser from grabbing the assets directly from CF, causing a 404. I've tried setting routes manually, but it seems like the middleware always runs no matter what I try.

**Comment by yusukebe** [comment]
@bruceharrison1984 

How about this?

```ts
app.get(
  '/static/*',
  (c) => {
    return c.text('Foo')
  },
  async (c, next) => {
    c.finalized = true
    await next()
  }
)

// Skipped
app.use('*', async (c, next) => {
  await next()
  c.header('X-Message', 'Hello')
})
```

**Reply by bruceharrison1984** [reply]
It seems like Hono is still capturing the request rather than allowing it to flow through to Cloudflare to retrieve the assets. 

This is the request to my stylesheet, but you can see it 404s. Using an explicit `use` route rather than a catch-all `*` allows those `/static/*` requests to pass by Hono without it tampering with them at all. I'm thinking this may be an unavoidable scenario unless there is someway to write a route that matches everything *except* what the route is. So basically an exclusionary route, which AFAIK isn't a thing that exists anywhere.

![image](https://github.com/honojs/hono/assets/17836372/0f42b552-d862-4279-af84-b24fe9625272)

**Comment by bruceharrison1984** [comment]
I figured out the cause of my issue and posted it over on the other repo: https://github.com/yusukebe/sonik/issues/8
